### PR TITLE
fix: use Chromium instead of Chrome

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

Replace Chrome by Chromium in the RUM agent Docker images

## Why is it important?

There is no Chrome binary for ARM

## Related issues
Closes https://github.com/elastic/apm-integration-testing/issues/1466

### CODE REVIEW

File: docker/opbeans/frontend/Dockerfile [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
-	gnupg \
-	--no-install-recommends \
-	&& curl -sSL https://dl.google.com/linux/linux_signing_key.pub | apt-key add - \
-	&& echo "deb [arch=amd64] https://dl.google.com/linux/chrome/deb/ stable main" > /etc/apt/sources.list.d/google-chrome.list \
-	&& apt-get -qq update && apt-get -qq install -y \
-	google-chrome-unstable \
-	--no-install-recommends \
+FROM node:16-slim
+
+ARG TARGETARCH=amd64
```
Comment (kuisathaverat) [inline docker/opbeans/frontend/Dockerfile:3]:
```suggestion
```

File: docker/opbeans/rum/Dockerfile [hunk 2]
```diff
@@ -1,21 +1,27 @@
-FROM node:12-slim
+FROM node:16-slim
 
 # Install latest chrome dev package and fonts to support major charsets (Chinese, Japanese, Arabic, Hebrew, Thai and a few others)
 # Note: this installs the necessary libs to make the bundled version of Chromium that Puppeteer
-# installs, work.
+# installs, work
+ARG TARGETARCH=amd64
```
Comment (kuisathaverat) [inline docker/opbeans/rum/Dockerfile:6]:
```suggestion
```

**Comment by dgieselaar** [comment]
I'm getting an error when building the `opbeans-rum` image:

```
Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
Building opbeans-rum
[+] Building 2.3s (6/16)                                                                                                                                                                                                                                                                              
 => [internal] load build definition from Dockerfile                                                                                                                                                                                                                                             0.0s
 => => transferring dockerfile: 2.33kB                                                                                                                                                                                                                                                           0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                                                                                0.0s
 => => transferring context: 2B                                                                                                                                                                                                                                                                  0.0s
 => [internal] load metadata for docker.io/library/node:16-slim                                                                                                                                                                                                                                  1.8s
 => CANCELED [ 1/11] FROM docker.io/library/node:16-slim@sha256:3c8acd4934617f60dad7e4cc941faa064aa5a14da437dc156bdcad9d4a67bc4e                                                                                                                                                                 0.3s
 => => resolve docker.io/library/node:16-slim@sha256:3c8acd4934617f60dad7e4cc941faa064aa5a14da437dc156bdcad9d4a67bc4e                                                                                                                                                                            0.0s
 => => sha256:4931fb8ba4223cb35034141162105ee5482122692b2931eb69eec912ce64606d 2.10MB / 25.91MB                                                                                                                                                                                                  0.3s
 => => sha256:9757dbba298c951be8e492425d90544397898ad5fde47d60665d494651b422a7 0B / 4.04kB                                                                                                                                                                                                       0.3s
 => => sha256:b435e038a2f6d99f52852489c6a2ca305e8537998f79609ef0d63e9f79db9599 0B / 34.67MB                                                                                                                                                                                                      0.3s
 => => sha256:3c8acd4934617f60dad7e4cc941faa064aa5a14da437dc156bdcad9d4a67bc4e 1.21kB / 1.21kB                                                                                                                                                                                                   0.0s
 => => sha256:8ef5ddf010addbee31a0988087391fac684e62ed471076f60b8ce9709037f282 1.37kB / 1.37kB                                                                                                                                                                                                   0.0s
 => => sha256:4341e43dd230da90b8ad912e81b8b346949419370bae02dce0ff5f2d2d8bfa50 7.05kB / 7.05kB                                                                                                                                                                                                   0.0s
 => ERROR https://github.com/Yelp/dumb-init/releases/download/v1.2.0/dumb-init_1.2.0_                                                                                                                                                                                                            0.3s
 => [internal] load build context                                                                                                                                                                                                                                                                0.0s
 => => transferring context: 66.36kB                                                                                                                                                                                                                                                             0.0s
------
 > https://github.com/Yelp/dumb-init/releases/download/v1.2.0/dumb-init_1.2.0_:
------
failed to load cache key: invalid response status 404
ERROR: Service 'opbeans-rum' failed to build : Build failed
Traceback (most recent call last):
  File "/Users/dariogieselaar/dev/apm-integration-testing/scripts/compose.py", line 31, in <module>
    main()
  File "/Users/dariogieselaar/dev/apm-integration-testing/scripts/compose.py", line 17, in main
    setup()
  File "/Users/dariogieselaar/dev/apm-integration-testing/scripts/modules/cli.py", line 212, in __call__
    self.args.func()
  File "/Users/dariogieselaar/dev/apm-integration-testing/scripts/modules/cli.py", line 589, in start_handler
    self.build_start_handler("start")
  File "/Users/dariogieselaar/dev/apm-integration-testing/scripts/modules/cli.py", line 758, in build_start_handler
    self.run_docker_compose_process(docker_compose_build + build_services)
  File "/Users/dariogieselaar/dev/apm-integration-testing/scripts/modules/cli.py", line 475, in run_docker_compose_process
    subprocess.check_call(docker_compose_cmd)
  File "/opt/homebrew/Cellar/python@3.9/3.9.10/Frameworks/Python.framework/Versions/3.9/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['docker-compose', '-f', '/Users/dariogieselaar/dev/apm-integration-testing/docker-compose.yml', 'build', '--pull', 'opbeans-rum', 'opbeans-ruby', 'opbeans-dotnet', 'opbeans-python', 'opbeans-node', 'opbeans-go', 'opbeans-java']' returned non-zero exit status 1.
```

**Review by cachedout — APPROVED** [review]

**Comment by kuisathaverat** [comment]
@dgieselaar my fault, I have removed the TARGETARCH directly on the PR, and it is needed, now should work

**Review by v1v — APPROVED** [review]

**Comment by dgieselaar** [comment]
The container starts, but is unhealthy, and ultimately `compose.py start` fails. Here's the error from the container:

```
2022-06-14T12:44:07: PM2 log: Launching in no daemon mode

2022-06-14T12:44:07: PM2 log: App [worker:0] starting in -cluster mode-

2022-06-14T12:44:07: PM2 log: App [worker:0] online

2022-06-14T12:46:05: PM2 log: Launching in no daemon mode

2022-06-14T12:46:05: PM2 log: App [worker:0] starting in -cluster mode-

2022-06-14T12:46:05: PM2 log: App [worker:0] online

Error: Failed to launch chrome!

qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory

TROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md

    at onClose (/home/pptruser/node_modules/puppeteer/lib/Launcher.js:348:14)

    at Interface.<anonymous> (/home/pptruser/node_modules/puppeteer/lib/Launcher.js:337:50)

    at Interface.emit (node:events:539:35)

    at Interface.close (node:readline:586:8)

    at Socket.onend (node:readline:277:10)

    at Socket.emit (node:events:539:35)

    at endReadableNT (node:internal/streams/readable:1345:12)

    at processTicksAndRejections (node:internal/process/task_queues:83:21)

Error: Failed to launch chrome!

qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory

TROUBLESHOOTING: https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md

    at onClose (/home/pptruser/node_modules/puppeteer/lib/Launcher.js:348:14)

    at Interface.<anonymous> (/home/pptruser/node_modules/puppeteer/lib/Launcher.js:337:50)

    at Interface.emit (node:events:539:35)

    at Interface.close (node:readline:586:8)

    at Socket.onend (node:readline:277:10)

    at Socket.emit (node:events:539:35)

    at endReadableNT (node:internal/streams/readable:1345:12)

    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```

**Comment by kuisathaverat** [comment]
that is an error on puppeteer, I am not sure we can run the Opnbeans RUM on ARM if puppeteer does not support that architecture. I will try to build the container for ARM locally and run it on QEMU, because I do not understand why is trying to find an x86_64 library and why is running on QEMU should build the native Docker image and run natively

**Comment by kuisathaverat** [comment]
I think is downloading the x86_64 versions of chrome https://github.com/puppeteer/puppeteer/issues/7740#issuecomment-1081225615

**Comment by kuisathaverat** [comment]
@dgieselaar  I hit a segmentfault but I am not sure if it could be related to run the Docker image in QEMU, Could you test the latest changes I have made
```
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
2022-06-16T16:07:14: PM2 log: Launching in no daemon mode
2022-06-16T16:07:15: PM2 log: App [worker:0] starting in -cluster mode-
2022-06-16T16:07:16: PM2 log: App [worker:0] online
Error: Failed to launch the browser process!
[0616/160727.153040:ERROR:stack_trace_posix.cc(668)] Failed to parse the contents of /proc/self/maps
clone: Invalid argument
[0616/160728.550026:FATAL:zygote_host_impl_linux.cc(190)] Check failed: ReceiveFixedMessage(fds[0], kZygoteBootMessage, sizeof(kZygoteBootMessage), &boot_pid). 
#0 0x005504f7c794 <unknown>
#1 0x005504eeb79c <unknown>
#2 0x005504efefc8 <unknown>
#3 0x005504eff848 <unknown>
#4 0x005503aa1f58 <unknown>
#5 0x005504ec7ec8 <unknown>
#6 0x005503204df8 <unknown>
#7 0x0055032052bc <unknown>
#8 0x005504ec7568 <unknown>
#9 0x005504ec5d98 <unknown>
#10 0x005504ec5eb4 <unknown>
#11 0x005504ee501c <unknown>
#12 0x005504ee4dd4 <unknown>
#13 0x0055029d819c ChromeMain
#14 0x00550f4fbd24 __libc_start_main
#15 0x0055029d8034 <unknown>
Received signal 6
#0 0x005504f7c794 <unknown>
#1 0x005504eeb79c <unknown>
#2 0x005504f7c75c <unknown>
#3 0x00550b862000 <unknown>
#4 0x00550f50d714 gsignal
#5 0x00550f4fb8e8 abort
#6 0x005504f7b850 <unknown>
#7 0x005504eff358 <unknown>
#8 0x005504eff848 <unknown>
#9 0x005503aa1f58 <unknown>
#10 0x005504ec7ec8 <unknown>
#11 0x005503204df8 <unknown>
#12 0x0055032052bc <unknown>
#13 0x005504ec7568 <unknown>
#14 0x005504ec5d98 <unknown>
#15 0x005504ec5eb4 <unknown>
#16 0x005504ee501c <unknown>
#17 0x005504ee4dd4 <unknown>
#18 0x0055029d819c ChromeMain
#19 0x00550f4fbd24 __libc_start_main
#20 0x0055029d8034 <unknown>
[end of stack trace]
Calling _exit(1). Core file will not be generated.
TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md
    at onClose (/home/pptruser/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:274:20)
    at ChildProcess.<anonymous> (/home/pptruser/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:268:79)
    at ChildProcess.emit (node:events:539:35)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:291:12)
```

**Comment by kuisathaverat** [comment]
I have checking on a M1 and it works on privileged mode, I have to check why and fix it

```
docker run --privileged -it docker.elastic.co/observability-ci/opbeans-rum-arm64:latest
```

**Comment by kuisathaverat** [comment]
```
> docker run -it  docker.elastic.co/observability-ci/opbeans-rum-arm64:latest
WARNING: The requested image's platform (linux/arm64) does not match the detected host platform (linux/amd64) and no specific platform was requested
2022-06-23T11:46:35: PM2 log: Launching in no daemon mode
2022-06-23T11:46:36: PM2 log: App [worker:0] starting in -cluster mode-
2022-06-23T11:46:36: PM2 log: App [worker:0] online
Error: Protocol error (Target.createTarget): Target closed.
    at /home/pptruser/node_modules/puppeteer/lib/Connection.js:74:56
    at new Promise (<anonymous>)
    at Connection.send (/home/pptruser/node_modules/puppeteer/lib/Connection.js:73:12)
    at Browser._createPageInContext (/home/pptruser/node_modules/puppeteer/lib/Browser.js:174:47)
    at BrowserContext.newPage (/home/pptruser/node_modules/puppeteer/lib/Browser.js:367:26)
    at Browser.newPage (/home/pptruser/node_modules/puppeteer/lib/Browser.js:166:33)
    at Browser.<anonymous> (/home/pptruser/node_modules/puppeteer/lib/helper.js:112:23)
    at run (/home/pptruser/tasks.js:16:30)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

```

**Comment by kuisathaverat** [comment]
The current status is that the Docker image work on `Linux/amd64` perfectly fine, and starts on `Linux/arm64` but does not work really, Chromium stuck on await browser.newPage()

**Pull request merged**